packages/server/src/models/shelf.entity.ts

```diff
@@ -0,0 +1,28 @@
+import { Column, Entity, PrimaryGeneratedColumn, PrimaryColumn } from 'typeorm';
+
+@Entity()
+export class Shelf {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column()
+  name: string;
+
+  @Column()
+  description: string;
+
+  @Column()
+  location: string;
+
+  @Column()
+  pictureUrl: string;
+}
+
+@Entity()
+export class ShelfFlowerOrdering {
```

Comment:
I would suggest to move it in a separate file

Reply:
And in next files

Reply:
Since I'm sharing the `ShelfDuties` abstract class in both `UserShelfDuties` and `UserShelfInvitation according to the [entity-inheritance](https://typeorm.io/#/entities/entity-inheritance) technique, I though it would make sense to keep them together, compact and obvious.

Also I don't think we must strictly stick to the rule of one entity per file.  Like here we have several entities that are inter-connected, all of them are about relation between a `user` and a `shelf`, so I thought keeping them in a single file makes sense for clarity. The file name itself, `user-shelf.entity`, might be a bit confusing but it rather names a related group of entities, not a single entity.

Having to duplicate field or export/export them just to have individual 1-2-1 file-to-entity relations would soon become overwhelming, and we would lose the clarity we're aiming for when separting them in the first place.

Reply:
I don't insist, guys, please share your thoughts on this. I'll do as more comfortable for the team.

Reply:
Why do we need ShelfFlowerOrdering and UserShelfOrdering ?

Reply:
`ShelfFlowerOrdering` is the ordering of a flower on a shelf (if we decide to add moving them with drag-n-drop, for instance, if we have time; each flower can only have one position on a particular shelf).
`UserShelfOrdering` is the order of shelves in the users interface (each user might want to sort shelves differently in his own interface).

Reply:
Fixed